### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by Misio12320** [Pokechu22/WorldDownloader#97]
Getting crash everytime when I join my server ver 4.0.3.1

```
---- Minecraft Crash Report ----
// Oops.

Time: 9/27/18 3:56 PM
Description: Ticking entity

java.lang.ClassCastException: java.util.Optional cannot be cast to java.lang.Integer
	at aoi.q(SourceFile:136)
	at aoi.b(SourceFile:121)
	at aoi.R_(SourceFile:103)
	at axs.a(SourceFile:1240)
	at axs.g(SourceFile:1215)
	at axs.o_(SourceFile:1097)
	at cfi.m(SourceFile:1277)
	at cfi.c(SourceFile:818)
	at cfi.a(SourceFile:392)
	at net.minecraft.client.main.Main.main(SourceFile:144)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- World Downloader Mod - Core --
Details:
	WDL version: 4.0.3.1
	Minecraft version: Minecraft 1.13-1.13.1 PH (MultiMC5/WorldDownloader-4.0.3.1/release)
	Expected version: 1.13-1.13.1 PH
	Protocol version: 401
	Data version: 1519
	File location: C:\Users\<USERNAME>\Desktop\MultiMC\instances\1.13\.minecraft\bin\minecraft.jar

-- World Downloader Mod - Extensions --
Details:
	Number loaded: 2
	Hologram: Id: Hologram
Version: 2.0
Display name: Hologram support
Main author: Pokechu22
Description: 
Provides basic support for disabling holograms.
Main class: wdl.HologramHandler
Containing file: C:\Users\<USERNAME>\Desktop\MultiMC\instances\1.13\.minecraft\bin\minecraft.jar
Implemented interfaces (2)
0: wdl.api.IEntityManager
1: wdl.api.IWDLModDescripted
Superclass: java.lang.Object
Classloader: sun.misc.Launcher$AppClassLoader@5c647e05 (sun.misc.Launcher$AppClassLoader)
Annotations (0)

	EntityRealigner: Id: EntityRealigner
Version: 1.0
Display name: Entity realigner
Main author: Pokechu22
Description: 
Realigns entities to their serverside position to deal with entities that drift clientside (for example, boats).
Main class: wdl.EntityRealigner
Containing file: C:\Users\<USERNAME>\Desktop\MultiMC\instances\1.13\.minecraft\bin\minecraft.jar
Implemented interfaces (2)
0: wdl.api.IEntityEditor
1: wdl.api.IWDLModDescripted
Superclass: java.lang.Object
Classloader: sun.misc.Launcher$AppClassLoader@5c647e05 (sun.misc.Launcher$AppClassLoader)
Annotations (0)


-- World Downloader Mod - State --
Details:
	minecraft: cfi@1ed43328
	worldClient: cqv@23216e17
	networkManager: hw@49842c0d
	thePlayer: csy['Sebastian_Z_GTA'/179780, l='MpServer', x=8.50, y=65.00, z=8.50]
	windowContainer: aqi@679864e0
	lastClickedBlock: ~~NULL~~
	lastEntity: ~~NULL~~
	saveHandler: ~~NULL~~
	chunkLoader: ~~NULL~~
	newTileEntities: {}
	newEntities: {}
	newMapDatas: {}
	downloading: false
	isMultiworld: true
	propsFound: true
	startOnChange: false
	overrideLastModifiedCheck: false
	saving: false
	worldLoadingDeferred: false
	worldName: 
	baseFolderName: teraxxx_aternos_me
	baseProps: wdl.config.Configuration@629e9957
	worldProps: wdl.config.Configuration@5decbb5f
	globalProps: wdl.config.Configuration@75952723
	defaultProps: wdl.config.DefaultConfiguration@510ee215

-- World Downloader Mod - Base properties --
Details:
	LinkedWorlds: |swiat|swiat

-- World Downloader Mod - World properties --
Details:
	-: empty

-- World Downloader Mod - Global properties --
Details:
	TutorialShown: true
	UpdateETag: "a2b6aea0be264735facbe8a5f8204fe3"

-- Entity being ticked --
Details:
	Entity Type: minecraft:arrow (aoi)
	Entity ID: 179785
	Entity Name: Strzała
	Entity's Exact location: -222.38, 98.38, 221.05
	Entity's Block location: World: (-223,98,221), Chunk: (at 1,6,13 in -14,13; contains blocks -224,0,208 to -209,255,223), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Entity's Momentum: -0.01, -0.05, -0.05
	Entity's Passengers: []
	Entity's Vehicle: ~~ERROR~~ NullPointerException: null

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [csy['Sebastian_Z_GTA'/179780, l='MpServer', x=8.50, y=65.00, z=8.50]]
	Chunk stats: MultiplayerChunkCache: 341, 341
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: {}
	Level spawn location: World: (8,64,8), Chunk: (at 8,4,8 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 11 game time, 11 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Forced entities: 35 total; [csy['Sebastian_Z_GTA'/179780, l='MpServer', x=8.50, y=65.00, z=8.50], alx['Ramka na przedmiot'/179782, l='MpServer', x=-223.50, y=98.50, z=212.03], akm[''/516, l='MpServer', x=-227.70, y=93.00, z=239.70], alx['Ramka na przedmiot'/179783, l='MpServer', x=-222.50, y=98.50, z=212.03], anx['Osadnik'/517, l='MpServer', x=-234.50, y=106.00, z=238.69], anx['Osadnik'/518, l='MpServer', x=-236.54, y=106.00, z=236.40], anx['Osadnik'/179781, l='MpServer', x=-223.30, y=102.00, z=218.07], anx['Osadnik'/519, l='MpServer', x=-234.55, y=106.00, z=234.39], anx['Osadnik'/179786, l='MpServer', x=-215.38, y=135.00, z=213.51], anx['Osadnik'/520, l='MpServer', x=-232.49, y=106.00, z=236.45], anx['Osadnik'/179787, l='MpServer', x=-214.70, y=135.00, z=211.59], alx['Ramka na przedmiot'/179784, l='MpServer', x=-221.03, y=97.50, z=213.50], alv[''/1034, l='MpServer', x=-211.13, y=97.90, z=270.24], aoi['Strzała'/179785, l='MpServer', x=-222.38, y=98.38, z=221.05], alx['Ramka na przedmiot'/179790, l='MpServer', x=-254.97, y=99.50, z=223.50], anx['Osadnik'/179791, l='MpServer', x=-218.75, y=101.06, z=188.55], akb['Owca'/179788, l='MpServer', x=-228.55, y=93.43, z=222.45], anx['Osadnik'/179789, l='MpServer', x=-218.54, y=99.06, z=196.40], aki['Wilk'/179794, l='MpServer', x=-260.27, y=64.00, z=204.47], aki['Wilk'/179795, l='MpServer', x=-258.39, y=64.00, z=207.53], anx['Osadnik'/179792, l='MpServer', x=-227.45, y=101.06, z=188.71], anm['Pająk'/179793, l='MpServer', x=-184.50, y=64.00, z=203.50], akg[''/179796, l='MpServer', x=-197.16, y=63.00, z=171.15], anm['Pająk'/179797, l='MpServer', x=-265.31, y=50.00, z=177.55], anj['Szkielet'/179621, l='MpServer', x=-217.50, y=96.00, z=255.50], anj['Szkielet'/179754, l='MpServer', x=-243.50, y=11.00, z=236.50], anj['Szkielet'/179755, l='MpServer', x=-242.50, y=11.00, z=236.50], aki['Wilk'/430, l='MpServer', x=-258.20, y=96.00, z=235.80], anx['Osadnik'/431, l='MpServer', x=-270.07, y=100.00, z=229.84], anx['Osadnik'/434, l='MpServer', x=-263.51, y=100.50, z=225.41], anx['Osadnik'/435, l='MpServer', x=-269.59, y=100.00, z=224.70], anx['Osadnik'/436, l='MpServer', x=-269.30, y=100.00, z=229.07], anj['Szkielet'/570, l='MpServer', x=-219.65, y=96.00, z=228.76], ajn['Kura'/571, l='MpServer', x=-212.32, y=96.00, z=262.88], anm['Pająk'/574, l='MpServer', x=-195.43, y=96.00, z=231.92]]
	Retry entities: 0 total; []
	Server brand: Spigot
	Server type: Non-integrated multiplayer server

-- System Details --
Details:
	Minecraft Version: 1.13
	Operating System: Windows 7 (amd64) version 6.1
	Java Version: 1.8.0_181, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 428317056 bytes (408 MB) / 701497344 bytes (669 MB) up to 954728448 bytes (910 MB)
	JVM Flags: 3 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xms512m -Xmx1024m
	Launched Version: MultiMC5
	LWJGL: 3.1.6 build 14
	OpenGL: GeForce GT 635M/PCIe/SSE2 GL version 4.6.0 NVIDIA 391.35, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'WorldDownloader-4.0.3.1'
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: Polski (Polska)
	Profiler Position: N/A (disabled)
	CPU: 4x Intel(R) Core(TM) i5-3230M CPU @ 2.60GHz
```

**Comment by Misio12320** [comment]
Now I cannot join because stupid sheep just making bee and crashing

**Comment by Misio12320** [comment]
[15:59:00] [Client thread/WARN]: Exception loading blockstate definition: minecraft:blockstates/dead_tube_coral.json: java.io.FileNotFoundException: minecraft:blockstates/dead_tube_coral.json
[15:59:00] [Client thread/WARN]: Exception loading blockstate definition: minecraft:blockstates/dead_brain_coral.json: java.io.FileNotFoundException: minecraft:blockstates/dead_brain_coral.json
[15:59:00] [Client thread/WARN]: Exception loading blockstate definition: minecraft:blockstates/dead_bubble_coral.json: java.io.FileNotFoundException: minecraft:blockstates/dead_bubble_coral.json
[15:59:00] [Client thread/WARN]: Exception loading blockstate definition: minecraft:blockstates/dead_fire_coral.json: java.io.FileNotFoundException: minecraft:blockstates/dead_fire_coral.json
[15:59:00] [Client thread/WARN]: Exception loading blockstate definition: minecraft:blockstates/dead_horn_coral.json: java.io.FileNotFoundException: minecraft:blockstates/dead_horn_coral.json
[15:59:01] [Client thread/WARN]: Unable to load model: 'minecraft:dead_brain_coral#inventory' referenced from: minecraft:dead_brain_coral#inventory: java.io.FileNotFoundException: minecraft:models/item/dead_brain_coral.json
[15:59:01] [Client thread/WARN]: Unable to load model: 'minecraft:dead_bubble_coral#inventory' referenced from: minecraft:dead_bubble_coral#inventory: java.io.FileNotFoundException: minecraft:models/item/dead_bubble_coral.json
[15:59:01] [Client thread/WARN]: Unable to load model: 'minecraft:dead_fire_coral#inventory' referenced from: minecraft:dead_fire_coral#inventory: java.io.FileNotFoundException: minecraft:models/item/dead_fire_coral.json
[15:59:01] [Client thread/WARN]: Unable to load model: 'minecraft:dead_horn_coral#inventory' referenced from: minecraft:dead_horn_coral#inventory: java.io.FileNotFoundException: minecraft:models/item/dead_horn_coral.json
[15:59:01] [Client thread/WARN]: Unable to load model: 'minecraft:dead_tube_coral#inventory' referenced from: minecraft:dead_tube_coral#inventory: java.io.FileNotFoundException: minecraft:models/item/dead_tube_coral.json

**Comment by Misio12320** [comment]
Or make a seed on 1.13.1 which makes empty void after chunks not loaded... Hm I Will see tomorrow more options what can I do with wdl

**Comment by Pokechu22** [comment]
The crash seems to be caused by messed up arrow entity metadata -- it looks like I didn't account for the [added shooter UUID field](https://wiki.vg/Protocol_History#18w30a) from 18w30a.  The game will crash whenever an arrow (or a trident) is loaded, from what I can tell.  It'll be fixed in the next release, which will be an _actual_ 1.13.1 release.

The "Exception loading blockstate definition" messages are expected, and just a side effect of how I implemented 1.13.1 from 1.13.  It'll result in there being no texture for those blocks, but no other issues.

I don't understand what you mean by "stupid sheep just making bee and crashing".  If you mean the sheep making a sound, in English that's written as [baa](https://en.wiktionary.org/wiki/baa) (I know other languages use different terms).  However, I'm not sure what that has to do anything, unless you're saying you hear the sheep and then crash.

Mess around with the options, the void chunks should already be possible.

**Comment by Misio12320** [comment]
>
> I don't understand what you mean by "stupid sheep just making bee and crashing". If you mean the sheep making a sound, in English that's written as [baa](https://en.wiktionary.org/wiki/baa) (I know other languages use different terms). However, I'm not sure what that has to do anything, unless you're saying you hear the sheep and then crash.
> 
> Mess around with the options, the void chunks should already be possible.

Lmfao I was thinking if the sheep crashed my game well but that was stress fast write etc... Hmm yes this empty void lock to dont get new chunks when flying around etc...

**Comment by Misio12320** [comment]
And will @Pokechu22 the download work when I set my server on 1.13? Well the chests works congrats lol. And the chests are 100% corrent any items more?

**Comment by Pokechu22** [comment]
Yes, it'll work fine with 1.13; the only issue is on 1.13.1.

Chests should be 100% correct now (v4.0.3.1).

**Comment by Misio12320** [comment]
When the 1.13.1 will release hm? Well nice  my English fix lmfao correct* 🤔

**Comment by Misio12320** [comment]
Is it possible to download everything inside the border with filled chunks?

**Comment by Pokechu22** [comment]
This has been fixed as of 4.0.3.2, which is a full 1.13.1 release.
